rust_src/src/frames.rs

```diff
@@ -683,6 +685,43 @@ pub fn frame_face_alist(frame: LispFrameLiveOrSelected) -> LispObject {
     frame_ref.face_alist
 }
 
+/// Update the display_time slot of the buffers shown in WINDOW and all its descendants.
+fn make_frame_visible_1(mut window: LispWindowRef) {
+    if window.is_null() {
+        return;
+    }
+    loop {
```

Comment:
Instead of a loop, let's implement an Iterator for LispWindowRef that knows how to walk the `next` window attribute. Look at LispSymbolRef's iterator as an example.

Reply:
Yup, you are right. I made this change in my latest commit.